### DISCUSSION TITLE
How to keep LuckyPerms ranks after using /premium

### DISCUSSION THREAD

**Discussion by MrAltik** [games647/FastLogin#777] [Q&A] [answered]
Hello,
I am kinda loosing my mind.
I cant find a way for players to keep their ranks after using /premium and that apply to inventories and possitions on others server on my bungeecord server.

I saw and tried on many servers using /premium with FastLogin and those servers keeps all player data from other plugins like LuckyPerms, invetories & positions and are still able to show also player heads in the tab after using /premium.

So they have "premiumUuid: true", cuz it wouldnt show players heads in the tab right?
How is it possible they have premium uuids and dont loose any player data that they had on cracked?

Please help me I am really loosing my mind over this, I was searching for answer so badly.

**Comment by TuxCoding** [comment, marked as answer]
>So they have "premiumUuid: true", cuz it wouldnt show players heads in the tab right?

No this option is independent from that. The client shows the head based on if it's an onlinemode connection. 

>How is it possible they have premium uuids and dont loose any player data that they had on cracked?

The plugin needs a way to convert the data. FastLogin doesn't integrate a converter, because it would be necessary for every plugin individually. However you can disable the premiumUUID and keep the old data

**Reply by MrAltik** [reply]
Hello,
thanks for answer! Could I ask one more?
If I set "premiumUuid: false" and enable "nameChangeCheck: true" and player changes its name, does the same offline uuid that player has stays after name change or is it entirely new offline uuid for new name change?

You know so the player wont loose anything with offline uuid even tho having premium account if they change their name.

**Reply by TuxCoding** [reply]
No the UUID changes too, because it depends on the player name in offline mode.
